src/todo_collection.js

```diff
@@ -0,0 +1,186 @@
+              <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
+              <path d="M12 12m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0" />
+              <path d="M12 19m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0" />
+              <path d="M12 5m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0" />
+            </svg>
+          `;
+          task.appendChild(svg);
+          this.todoList.appendChild(line);
+          this.todoList.appendChild(task);
+
+          /* Add event listener to the checkbox */
+
+          checkInput.addEventListener('change', () => {
+            if (this.todoData[i].completed === false) {
+              this.todoData[i].completed = true;
+              localStorage.setItem('todoData', JSON.stringify(this.todoData));
+              item.classList.add('completed');
+            } else {
+              this.todoData[i].completed = false;
+              localStorage.setItem('todoData', JSON.stringify(this.todoData));
+              item.classList.remove('completed');
+            }
+          });
+
+          item.addEventListener('click', () => {
+            const inputList = Array.from(document.querySelectorAll('.editInput'));
+            const editText = Array.from(document.querySelectorAll('.editText'));
+            const bgInput = Array.from(document.querySelectorAll('.main__item'));
+            const svgBtn = Array.from(document.querySelectorAll('.svg'));
+            if (editInput.classList.contains('hidden')) {
+              const valInput = inputList.find((item) => !item.classList.contains('hidden'));
+              const valText = editText.find((item) => item.classList.contains('hidden'));
+              const valBg = bgInput.find((item) => item.classList.contains('bg-yellow'));
+              const valSvg = svgBtn.find((item) => item.classList.contains('trash'));
+              const svgList = document.querySelectorAll('.svg');
+
+              if (valInput) {
+                valInput.classList.add('hidden');
+                valText.classList.remove('hidden');
+                valBg.classList.remove('bg-yellow');
+                valSvg.classList.remove('trash');
+              }
+
+              editInput.classList.remove('hidden');
+              editInput.focus();
+              itemText.classList.add('hidden');
+              task.classList.add('bg-yellow');
+              svg.classList.add('trash');
+
+              const svgTrash = document.querySelector('.trash');
+
+              svgTrash.addEventListener('click', () => {
+                this.todoData.splice(i, 1);
+                for (let j = i; j < this.todoData.length; j += 1) this.todoData[j].index = j;
+                localStorage.setItem('todoData', JSON.stringify(this.todoData));
+                this.render();
+              });
```

Comment:
This seems to be the same code written in the delTask Function, please try to reuse codes so as to have efficient and maintainable code base

Reply:
solved